src/djinterop/enginelibrary/el_track_impl.cpp

```diff
@@ -201,6 +201,12 @@ overview_waveform_data el_track_impl::get_overview_waveform_data()
 
 void el_track_impl::set_overview_waveform_data(overview_waveform_data data)
 {
+    for (auto&& entry : data.waveform)
+    {
+        entry.low.opacity = 255;
+        entry.mid.opacity = 255;
+        entry.high.opacity = 255;
+    }
```

Comment:
It is already the case that the opacity is set to a default value (`255`) in `overview_waveform_data::decode()`, because of this line: https://github.com/xsco/libdjinterop/blob/e159163e8ff02dc14a4f26a538ced9b0a483fdb6/include/djinterop/performance_data.hpp#L64

The check which I added in `el_track_impl.hpp` would fail if we don't write the same value (`255`).

Reply:
Thanks, this wasn't immediately obvious to me.  Perhaps we can add an explanatory comment?

```suggestion
    // As the overview waveform does not store opacity, it is defaulted to 255
    // when read back.  If we also set it to 255 here, we can apply a check in
    // `set_perfdata` that a round-trip encode/decode gives the same data.
    for (auto&& entry : data.waveform)
    {
        entry.low.opacity = 255;
        entry.mid.opacity = 255;
        entry.high.opacity = 255;
    }
```